lib/hanami/configuration/router.rb

```diff
@@ -1,50 +1,50 @@
 # frozen_string_literal: true
 
+require "dry/configurable"
+require_relative "../application/routing/resolver"
+
 module Hanami
   class Configuration
     # Hanami router configuration
     #
     # @since 2.0.0
     # @api private
     class Router
-      # @api private
-      # @since 2.0.0
-      attr_writer :routes
+      include Dry::Configurable
 
-      # @api private
-      # @since 2.0.0
-      attr_reader :routes
+      # Base configuration is provided so router config can include the `base_url`
+      attr_reader :base_configuration
+      private :base_configuration
 
       # @api private
       # @since 2.0.0
-      attr_writer :resolver
-
-      # @api private
-      # @since 2.0.0
-      def initialize(base_url, routes: DEFAULT_ROUTES)
-        @base_url = base_url
-        @routes = routes
+      def initialize(base_configuration)
+        @base_configuration = base_configuration
       end
 
-      # @api private
-      # @since 2.0.0
-      def resolver
-        @resolver ||= begin
-                        require_relative "../application/routing/resolver"
-                        Application::Routing::Resolver
-                      end
-      end
+      setting :routes_path, File.join("config", "routes")
+
+      setting :resolver, Application::Routing::Resolver
 
       # @api private
       # @since 2.0.0
       def options
-        { base_url: @base_url }
+        {base_url: base_configuration.base_url}
       end
 
-      # @api private
-      # @since 2.0.0
-      DEFAULT_ROUTES = File.join("config", "routes")
-      private_constant :DEFAULT_ROUTES
+      private
+
+      def method_missing(name, *args, &block)
```

Comment:
Can't we get rid of it if we make the settings with `reader: true` option?

Reply:
We need it for the writers still. I'd love to have dry-configurable support this kind of use case out of the box, since I've written this particular method_missing implementation about 10 or more times in my Hanami work so far.